### PR TITLE
fix: billing docs and added update and delete endpoints

### PR DESCRIPTION
**Pull request by kaleabendrias**
### What does the PR do?

#Closes: #1100 
This PR introduces new functionality for updating and deleting billing plans. It adds endpoints for modifying existing billing plans and removing them from the database. The `updateBillingPlan` method allows for partial updates to a billing plan, and the `deleteBillingPlan` method handles the deletion of a billing plan by ID. Additionally, Swagger API documentation has been enhanced with custom decorators for better reusability and cleaner code.

**How should this be manually tested?**

1. **Test the `updateBillingPlan` Endpoint:**
   - Send a `PATCH` request to `/billing-plans/:id` with a valid billing plan ID and an updated payload.
   - Verify that the billing plan is updated correctly in the database.
   - Test with an invalid billing plan ID to ensure that it returns a `404 Not Found` error.

2. **Test the `deleteBillingPlan` Endpoint:**
   - Send a `DELETE` request to `/billing-plans/:id` with a valid billing plan ID.
   - Confirm that the billing plan is removed from the database.
   - Test with an invalid billing plan ID to ensure that it returns a `404 Not Found` error.

3. **Verify Swagger Documentation:**
   - Access the Swagger documentation for the API.
   - Ensure that the `update` and `delete` endpoints are documented with the correct details and response types.

**Checklist of what you did:**
- Implemented `updateBillingPlan` method in the controller to allow partial updates of billing plans.
- Implemented `deleteBillingPlan` method in the controller to enable deletion of billing plans.
- Updated the `BillingPlanService` to include logic for updating and deleting billing plans.
- Created custom Swagger decorators for `PATCH` and `DELETE` endpoints to manage API documentation.
- Ensured error handling is in place for invalid billing plan IDs.

### CODE REVIEW

File: src/modules/billing-plans/billing-plan.service.ts [hunk 1]
```diff
@@ -63,4 +64,22 @@ export class BillingPlanService {
       data: plan,
     };
   }
+
+  async updateBillingPlan(id: string, updateBillingPlanDto: UpdateBillingPlanDto): Promise<BillingPlan> {
+    const billing_plan = await this.billingPlanRepository.findOneBy({ id });
+    console.log(billing_plan);
```
Comment (Cobidex) [inline src/modules/billing-plans/billing-plan.service.ts:70]: Remove console.log

File: src/modules/billing-plans/billing-plan.controller.ts [hunk 2]
```diff
@@ -1,42 +1,71 @@
...
-  @ApiOperation({ summary: 'Get single billing plan by ID' })
-  @ApiResponse({ status: 200, description: 'Billing plan retrieved successfully', type: BillingPlanDto })
-  @ApiResponse({ status: 400, description: 'Invalid billing plan ID' })
-  @ApiResponse({ status: 404, description: 'Billing plan not found' })
   async getSingleBillingPlan(@Param('id') id: string) {
     return this.billingPlanService.getSingleBillingPlan(id);
   }
+
+  @UseGuards(SuperAdminGuard)
+  @updateBillingPlan()
```
Comment (Cobidex) [inline src/modules/billing-plans/billing-plan.controller.ts:55]: Name this as uodateBillingPlanDoc so it's more descriptive

File: src/modules/billing-plans/billing-plan.controller.ts [hunk 3]
```diff
@@ -1,42 +1,71 @@
...
+  @Patch('/:id')
+  async updateBillingPlan(
+    @Param('id', new ParseUUIDPipe()) id: string,
+    @Body() updateBillingPlanDto: UpdateBillingPlanDto
+  ) {
+    return this.billingPlanService.updateBillingPlan(id, updateBillingPlanDto);
+  }
+
+  @UseGuards(SuperAdminGuard)
+  @deleteBillingPlan()
```
Comment (Cobidex) [inline src/modules/billing-plans/billing-plan.controller.ts:65]: Same here adjust the name

File: src/modules/billing-plans/billing-plan.service.ts (hunk 1 above)
Comment (kaleabendrias) [reply src/modules/billing-plans/billing-plan.service.ts]: @Cobidex i have resolved it.

File: src/modules/billing-plans/docs/billing-plan-docs.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+export function getAllBillingPlansDocs() {
+  return applyDecorators(
+    ApiOperation({ summary: 'Get all billing plans' }),
+    ApiResponse({ status: 200, description: 'Billing plans retrieved successfully.', type: [BillingPlanDto] }),
+    ApiResponse({ status: 404, description: 'No billing plans found.' })
+  );
+}
+
+export function getSingleBillingPlan() {
```
Comment (Cobidex) [inline src/modules/billing-plans/docs/billing-plan-docs.ts:24]: Rename

File: src/modules/billing-plans/docs/billing-plan-docs.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  return applyDecorators(
+    ApiOperation({ summary: 'Get single billing plan by ID' }),
+    ApiResponse({ status: 200, description: 'Billing plan retrieved successfully', type: BillingPlanDto }),
+    ApiResponse({ status: 400, description: 'Invalid billing plan ID' }),
+    ApiResponse({ status: 404, description: 'Billing plan not found' })
+  );
+}
+
+export function updateBillingPlan() {
```
Comment (Cobidex) [inline src/modules/billing-plans/docs/billing-plan-docs.ts:33]: Rename

File: src/modules/billing-plans/docs/billing-plan-docs.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  return applyDecorators(
+    ApiOperation({ summary: 'Update a billing plan by ID' }),
+    ApiResponse({ status: 200, description: 'Billing plan updated successfully.', type: BillingPlan }),
+    ApiResponse({ status: 404, description: 'Billing plan not found.' })
+  );
+}
+
+export function deleteBillingPlan() {
```
Comment (Cobidex) [inline src/modules/billing-plans/docs/billing-plan-docs.ts:41]: Rename

**Review by Cobidex — APPROVED** [review]

File: src/modules/billing-plans/billing-plan.controller.ts (hunk 2 above)
Comment (kaleabendrias) [reply src/modules/billing-plans/billing-plan.controller.ts]: done

File: src/modules/billing-plans/billing-plan.controller.ts (hunk 3 above)
Comment (kaleabendrias) [reply src/modules/billing-plans/billing-plan.controller.ts]: done

**Pull request merged**